.github/workflows/ci.yaml

```
@@ -72,3 +72,18 @@ jobs:
           token: ${{ secrets.CODECOV_TOKEN }}
           files: lcov.info
           fail_ci_if_error: true
+  test-metal:
+    name: Test Metal
+    runs-on: [self-hosted, macOS]
+    env:
+      CARGO_TERM_COLOR: always
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          components: clippy
+      - name: Install cargo-llvm-cov
+        uses: taiki-e/install-action@cargo-llvm-cov
+      - name: Run tests
+        run: cargo test -F metal
```

```suggestion
        run: cargo test -F metal -- metal
```
for executing only those in the `metal` module

---

Im not sure about that, I would like to run all test that use metal indirectly too

---

you're right! my bad